providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/FractionalTest.java

```diff
@@ -189,7 +189,49 @@ void invalidRuleSumNot100() throws JsonLogicEvaluationException {
         Object evaluate = fractional.evaluate(rule, data);
 
         // then
-        assertNull(evaluate);
+        assertEquals("blue", evaluate);
+    }
+
+    @Test
+    void invalidRuleSumlower100() throws JsonLogicEvaluationException {
+        // given
+        Fractional fractional = new Fractional();
+
+        /* Rule
+         *     [
+         *       [
+         *         "blue",
+         *         50
+         *       ],
+         *       [
+         *         "green",
+         *         30
+         *       ]
+         *     ]
+         * */
+
+        final List<Object> rule = new ArrayList<>();
+
+        final List<Object> bucket1 = new ArrayList<>();
+        bucket1.add("blue");
+        bucket1.add(50);
+
+        final List<Object> bucket2 = new ArrayList<>();
+        bucket2.add("green");
+        bucket2.add(70);
```

Comment:
nit - to match with the rule

```suggestion
        bucket2.add(30);
```